File: MyCucumberTest/src/page/LandingPage.java
package page;

import org.openqa.selenium.By;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import baseline.AutoContext;
import baseline.MyDriver;
import pageObject.PageLoader;
import pageObject.PageObject;

public class LandingPage implements PageObject {

	public MyDriver myDriver = AutoContext.myWebDriverTL.get();
	static final Logger logger = LoggerFactory.getLogger(LandingPage.class);
	private String menu = "//a[@id='menu']/span";
	
	public MenuPage clickMenu(){
		myDriver.findMyWebElement(By.xpath(menu));		
		myDriver.perform(By.xpath(menu));
		return PageLoader.load(MenuPage.class);
	}
}

File: CucumberRuby/MyTest/features/Page/LandingPage.rb
require 'capybara'
require 'capybara/cucumber'
require 'selenium-webdriver'
require 'site_prism'

class LandingPage < SitePrism::Page

  #element :menu, "a[id='menu']"
  element :menu, :xpath, "//a[@id='menu']"
 
  def click_menu
    #wait_until{ menu.visible? }
  	#menu.click 
  	execute_script("$(menu).click()")
  end
end
